supervisord.py

```diff
@@ -47,14 +48,29 @@ def check(self, instance):
             AgentCheck.UNKNOWN: 0
         }
 
-        # Report service checks and uptime for each process
+        proc_regex = instance.get('proc_regex')
         proc_names = instance.get('proc_names')
+        processes = server.supervisor.getAllProcessInfo()
+        monitored_processes = []
+
+        # default to all processes if no process filters are configured
+        if not proc_regex and not proc_names:
```

Comment:
Here you're not checking the length, but below you are checking the length of the arrays. This means that you could do:

```yaml
  proc_names:
  proc_regex:
```

And it wouldn't match anything (a corner case, but still probably not intended behavior).